crates/solver/src/settlement_access_list.rs

```diff
@@ -74,22 +100,81 @@ impl TenderlyApi {
             .json()
             .await
     }
+
+    async fn block_number(&self, network_id: String) -> reqwest::Result<BlockNumber> {
+        self.client
+            .get(format!(
+                "https://api.tenderly.co/api/v1/network/{}/block-number",
+                network_id
+            ))
+            .send()
+            .await?
+            .error_for_status()?
+            .json()
+            .await
+    }
+}
+
+#[async_trait::async_trait]
+impl AccessListEstimating for TenderlyApi {
+    async fn estimate_access_lists(
+        &self,
+        txs: &[TransactionBuilder<DynTransport>],
+    ) -> Vec<Result<AccessList>> {
+        futures::future::join_all(txs.iter().map(|tx| async {
+            let input = tx
+                .data
+                .clone()
+                .ok_or(anyhow!("transaction data does not exist"))?;
```

Comment:
ubernit: It would be slightly better to use `ok_or_else` here so that we don't build an error if we don't need it:
```suggestion
                .ok_or_else(|| anyhow!("transaction data does not exist"))?;
```

also, I _think_ `anyhow` has a `Context` implementation for `Option<_>` so that you can turn it into a `Result<_, anyhow::Error>` a little more succinctly:

```rust
use anyhow::Context as _;
```

```suggestion
                .context("transaction data does not exist")?;
```

Reply:
Cool, switched to `context`